tfe/resource_tfe_team_access_migrate_test.go

```diff
@@ -0,0 +1,27 @@
+package tfe
+
+func testResourceTfeTeamAccessStateDataV0() map[string]interface{} {
+	return map[string]interface{}{
+		"workspace_id": "hashicorp/workspace",
+	}
+}
+
+func testResourceTfeTeamAccessStateDataV1() map[string]interface{} {
+	return map[string]interface{}{
+		"workspace_id": "ws-123",
+	}
+}
+
+//TODO: how would I test this since I have to pass a tfe.Client as the meta? I don't want to mock out the entire client.
```

Comment:
Would it work if you passed the same meta that has tfe.Client that the other resource tests get access to (`testAccProvider.Meta().(*tfe.Client)` or `testAccProvider.Meta()`)?

Reply:
@beekus the client is build up using interfaces, so you can mock out only the parts that you would need for your tests.

Have a look at this example where I did the same for testing the remote backend: https://github.com/hashicorp/terraform/blob/master/backend/remote/testing.go#L97

Of course here I use a lot of the endpoints so I needed to mock out quite a bit, I assume you would need less things for testing this part...

Reply:
Awesome thanks. I think I got this sorted now.